bCNC/controllers/_GenericController.py

```diff
@@ -8,8 +8,8 @@
 import time
 import re
 
-STATUSPAT = re.compile(r"^<(\w*?),MPos:([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),WPos:([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),?(.*)>$")
-POSPAT	  = re.compile(r"^\[(...):([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*):?(\d*)\]$")
+STATUSPAT = re.compile(r"^<(\w*?),MPos:([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),WPos:([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),?(.*)>$")
+POSPAT	  = re.compile(r"^\[(...):([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*),([+\-]?\d*\.\d*):?(\d*)\]$")
```

Comment:
the new groups in the pattern are non-optional, aren't they? With my GRBL I get `[PRB:-356.0,-5.28,-118.3,0.0:1]`, i.e. only 4 axes. What do I miss?

Actually, even the old pattern with only 3 coordinates is broken with my 4 axes GRBL.

Reply:
Are you able to provide PR with fix please?

Reply:
https://github.com/vlachoudis/bCNC/pull/1441

Reply:
Not tested yet. 

Spent the afternoon hunting issues with z probing, giving wrong values. I hope this is was the root cause. Going to try now on real machine.